src/lib/validators.js

```diff
@@ -132,3 +133,26 @@ export const validateUnique = arr => {
 
 export const validateChild = ourShip => ship =>
   ourShip !== ob.sein(ship) && `This point is not a child of ${ourShip}.`;
+
+export const validatePsbt = base64 => {
+  try {
+    bitcoin.Psbt.fromBase64(base64);
+  } catch (e) {
+    return 'Invalid Partially Signed Bitcoin Transaction';
+  }
+};
+
+export const validateSignablePsbt = hd => base64 => {
+    const psbt = bitcoin.Psbt.fromBase64(base64);
+    return !psbt.data.inputs.some((input, idx) => {
```

Comment:
Instead of negating, might be simpler to remove the negation here, and say `|| 'No inputs were signed'` below? That feels like the more common pattern.

Reply:
The the "!test && error" pattern used throughout this file isn't my jam either, but that's what this file seems to be all about.

Reply:
+1 for trying to match the surrounding code, but I hereby give you permission to flaunt local convention and just do what seems most sane to you. (:

Reply:
nvm i'm wrong, it's not equivalent.